packages/iTwinUI-react/src/core/Tree/Tree.tsx

```diff
@@ -224,40 +245,173 @@ export const Tree = <T,>(props: TreeProps<T>) => {
     return [flatList, firstLevelNodes];
   }, [data, getNode]);
 
-  return (
-    <ul
-      className={cx('iui-tree', className)}
-      role='tree'
-      onKeyDown={handleKeyDown}
-      ref={treeRef}
-      tabIndex={0}
-      onFocus={() => {
-        const items = getFocusableNodes();
-        if (items.length > 0) {
-          items[focusedIndex.current]?.focus();
-        }
-      }}
-      {...rest}
-    >
-      {flatNodesList.map((flatNode) => (
+  const itemRenderer = React.useCallback(
+    (index: number) => {
+      const node = flatNodesList[index];
+      return (
         <TreeContext.Provider
-          key={flatNode.nodeProps.nodeId}
+          key={node.nodeProps.nodeId}
           value={{
-            nodeDepth: flatNode.depth,
-            subNodeIds: flatNode.subNodeIds,
+            nodeDepth: node.depth,
+            subNodeIds: node.subNodeIds,
             groupSize:
-              flatNode.depth === 0
+              node.depth === 0
                 ? firstLevelNodesList.length
-                : flatNode.parentNode?.subNodeIds?.length ?? 0,
-            indexInGroup: flatNode.indexInGroup,
-            parentNodeId: flatNode.parentNode?.nodeProps.nodeId,
+                : node.parentNode?.subNodeIds?.length ?? 0,
+            indexInGroup: node.indexInGroup,
+            parentNodeId: node.parentNode?.nodeProps.nodeId,
+            scrollToParent: node.parentNode
+              ? () => {
+                  const parentNodeId = node.parentNode?.nodeProps.nodeId;
+                  const parentNodeIndex = flatNodesList.findIndex(
+                    (n) => n.nodeProps.nodeId === parentNodeId,
+                  );
+                  setScrollToIndex(parentNodeIndex);
+                }
+              : undefined,
           }}
         >
-          {nodeRenderer(flatNode.nodeProps)}
+          {nodeRenderer(node.nodeProps)}
         </TreeContext.Provider>
-      ))}
-    </ul>
+      );
+    },
+    [firstLevelNodesList.length, flatNodesList, nodeRenderer],
+  );
+
+  const [scrollToIndex, setScrollToIndex] = React.useState<number>();
+  const flatNodesListRef = React.useRef<FlatNode<T>[]>(flatNodesList);
+  flatNodesListRef.current = flatNodesList;
+
+  React.useEffect(() => {
+    setTimeout(() => {
+      if (scrollToIndex !== undefined) {
+        const nodeId = flatNodesListRef.current[scrollToIndex].nodeProps.nodeId;
+        const nodeElement = treeRef.current?.ownerDocument.querySelector(
+          `#${nodeId}`,
+        ) as HTMLElement;
+        nodeElement?.focus();
+        // Need to reset that if navigating with mouse and keyboard,
+        // e.g. going to the previous node with keyboard
+        // and then clicking the next node with mouse.
+        setScrollToIndex(undefined);
```

Comment:
Not sure I wonderstand what this is really doing. I didn't notice any difference when I commented it out and tried going to previous node then clicking next node with mouse.

Reply:
Bad comment, changed it a bit. It was needed when using arrow left to go back to the parent node. So if you press arrow left and go to the parent node, then click with the mouse some child node and try pressing arrow left again, without this reset it would do nothing because `scrollToIndex` did not change.